react/features/toolbox/components/PresenterMuteButton.js

```diff
@@ -0,0 +1,176 @@
+// @flow
+
+import {
+    ACTION_SHORTCUT_TRIGGERED,
+    PRESENTER_MUTE,
+    createShortcutEvent,
+    createToolbarEvent,
+    sendAnalytics
+} from '../../analytics';
+import { setAudioOnly } from '../../base/audio-only';
+import { translate } from '../../base/i18n';
+import {
+    MEDIA_TYPE,
+    VIDEO_MUTISM_AUTHORITY,
+    setPresenterMuted
+} from '../../base/media';
+import { connect } from '../../base/redux';
+import { AbstractPresenterMuteButton } from '../../base/toolbox';
+import type { AbstractButtonProps } from '../../base/toolbox';
+import { isLocalTrackMuted } from '../../base/tracks';
+import UIEvents from '../../../../service/UI/UIEvents';
+
+declare var APP: Object;
+
+/**
+ * The type of the React {@code Component} props of {@link PresenterMuteButton}.
+ */
+type Props = AbstractButtonProps & {
+
+    /**
+     * Whether the current conference is in audio only mode or not.
+     */
+    _audioOnly: boolean,
+
+    /**
+     * Whether presenter video is currently muted or not.
+     */
+    _presenterMuted: boolean,
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function
+}
+
+/**
+ * Component that renders a toolbar button for toggling presenter mute.
+ *
+ * @extends AbstractPresenterMuteButton
+ */
+class PresenterMuteButton extends AbstractPresenterMuteButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.presentermute';
+    label = 'toolbar.presentermute';
+    tooltip = 'toolbar.presentermute';
+
+    /**
+     * Initializes a new {@code PresenterMuteButton} instance.
+     *
+     * @param {Props} props - The read-only React {@code Component} props with
+     * which the new instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+
+        // Bind event handlers so they are only bound once per instance.
+        this._onKeyboardShortcut = this._onKeyboardShortcut.bind(this);
+    }
+
+    /**
+     * Registers the keyboard shortcut that toggles the presenter muting.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        typeof APP === 'undefined'
+            || APP.keyboardshortcut.registerShortcut(
+                'V',
+                null,
+                this._onKeyboardShortcut,
+                'keyboardShortcuts.presenterMute');
+    }
+
+    /**
+     * Unregisters the keyboard shortcut that toggles the presenter muting.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        typeof APP === 'undefined'
+            || APP.keyboardshortcut.unregisterShortcut('V');
```

Comment:
Tangent (not part of actual code feedback): do you remember if keyboardshortcut needs to be on the global APP still? I was also wondering why I never bothered to just move it into a module that gets imported and does the APP truthy check.